src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -191,16 +192,32 @@ private Bundle normalizeExtras(Bundle extras) {
         return newExtras;
     }
 
+    private void updateBadge(Context context, String badge) {
+
+        int count = badge == null ? 0 : Integer.parseInt(badge);
```

Comment:
This would make it impossible to keep the badge count unchanged (eg. when it's not set)